cloudinit/net/network_manager.py

```diff
@@ -71,6 +71,60 @@ def _set_default(self, section, option, value):
         if not self.config.has_option(section, option):
             self.config[section][option] = value
 
+    def _config_option_is_set(self, section, option):
+        """
+        Checks if a config option is set. Returns True if it is,
+        else returns False.
+        """
+        return self.config.has_section(section) and self.config.has_option(
+            section, option
+        )
+
+    def _get_config_option(self, section, option):
+        """
+        Returns the value of a config option if its set,
+        else returns None.
+        """
+        if self._config_option_is_set(section, option):
+            return self.config[section][option]
+        else:
+            return None
+
+    def _change_set_config_option(self, section, option, value):
+        """
+        Overrides the value of a config option if its already set.
+        Else, if the config option is not set, it does nothing.
+        """
+        if self._config_option_is_set(section, option):
+            self.config[section][option] = value
+
+    def _set_mayfail_true_if_both_false_dhcp(self):
+        """
+        If for both ipv4 and ipv6, 'may-fail' is set to be False,
+        set it to True for both of them.
+        """
+        for family in ["ipv4", "ipv6"]:
```

Comment:
Can this be simplified to:
```python
        for family in ["ipv4", "ipv6"]:
            if self._get_config_option(family, "method") not in [
                "dhcp",
                "auto",
            ]:
                # if both v4 and v6 are not dhcp, do not do anything.
                return
```

We don't really care about the `may-fail` value, correct? Rather we care if both methods are using DHCP.

Reply:
> Can this be simplified to:
> 
> ```python
>         for family in ["ipv4", "ipv6"]:
>             if self._get_config_option(family, "method") not in [
>                 "dhcp",
>                 "auto",
>             ]:
>                 # if both v4 and v6 are not dhcp, do not do anything.
>                 return
> ```
> 
> We don't really care about the `may-fail` value, correct? Rather we care if both methods are using DHCP.

We do care about may-fail. If both methods are dhcp and in both cases may-fail is False then we would want to make may-fail as True. So both checks are needed.

Reply:
I understand that currently the way the code is written, "may-fail" is always set to False for dhcp but I want to have an explicit check to make it future proof and to show specific intention.